src/basic/dirent-util.c

```diff
@@ -19,15 +19,17 @@ int dirent_ensure_type(DIR *d, struct dirent *de) {
         if (fstatat(dirfd(d), de->d_name, &st, AT_SYMLINK_NOFOLLOW) < 0)
                 return -errno;
 
-        de->d_type =
-                S_ISREG(st.st_mode)  ? DT_REG  :
-                S_ISDIR(st.st_mode)  ? DT_DIR  :
-                S_ISLNK(st.st_mode)  ? DT_LNK  :
-                S_ISFIFO(st.st_mode) ? DT_FIFO :
-                S_ISSOCK(st.st_mode) ? DT_SOCK :
-                S_ISCHR(st.st_mode)  ? DT_CHR  :
-                S_ISBLK(st.st_mode)  ? DT_BLK  :
-                                       DT_UNKNOWN;
+        de->d_type = S_ISREG(st.st_mode) ?
+                DT_REG :
+                S_ISDIR(st.st_mode) ?
+                DT_DIR :
+                S_ISLNK(st.st_mode) ?
+                DT_LNK :
+                S_ISFIFO(st.st_mode) ?
+                DT_FIFO :
+                S_ISSOCK(st.st_mode) ?
+                DT_SOCK :
+                S_ISCHR(st.st_mode) ? DT_CHR : S_ISBLK(st.st_mode) ? DT_BLK : DT_UNKNOWN;
```

Comment:
hmm, this certainly makes it less readable... i much preferred the table-like alignment here...

Reply:
IMHO it would be better to avoid these multiple nested ternary operators completely. 
better to replace with if else or with a switch case:

```cpp
        if (S_ISREG(st.st_mode))
                de->d_type = DT_REG;
        else if (S_ISDIR(st.st_mode))
                de->d_type = DT_DIR;
        else if (S_ISLNK(st.st_mode))
                de->d_type = DT_LNK;
        else if (S_ISFIFO(st.st_mode))
                de->d_type = DT_FIFO;
        else if (S_ISSOCK(st.st_mode))
                de->d_type = DT_SOCK;
        else if (S_ISCHR(st.st_mode))
                de->d_type = DT_CHR;
        else if (S_ISBLK(st.st_mode))
                de->d_type = DT_BLK;
        else
                de->d_type = DT_UNKNOWN;
```